Dockerfile.in

```diff
@@ -5,6 +5,34 @@ RUN sed -ri 's!(httpredir|deb)\.debian\.org!mirrors.ocf.berkeley.edu!g; \
         s!security\.debian\.org!mirrors.ocf.berkeley.edu!g' \
         /etc/apt/sources.list
 
+# Install a few packages so that the OCF apt key can be added and so that any
+# pre-package installation commands work
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+    ca-certificates curl gnupg
+
+
+{% if ocf_apt_repo_dist %}
+# add ocf apt repo
+RUN curl https://apt.ocf.berkeley.edu/pubkey.gpg | apt-key add - && \
+    echo 'deb http://apt.ocf.berkeley.edu/ {{ocf_apt_repo_dist}} main' \
+      >> /etc/apt/sources.list
+{% endif %}
+
+{% if backport_dist %}
+COPY backports.pref /etc/apt/preferences.d
+RUN echo 'deb http://mirrors.ocf.berkeley.edu/debian/ {{backport_dist}} main' \
+      >> /etc/apt/sources.list
+{% if ocf_apt_repo_dist %}
+RUN echo 'deb http://apt.ocf.berkeley.edu/ {{backport_dist}} main' \
+      >> /etc/apt/sources.list
+{% endif %}
+{% endif %}
+
+# add deb-src entries
+RUN sed 's/^deb /deb-src /' /etc/apt/sources.list >> /etc/apt/sources.list
```

Comment:
do we actually need these in a dockerfile?

Reply:
Generally no, but it's nice if you want to compile something or package something in docker, which is often what I find it is used for. I don't think this really adds any overhead, and it was present before, so I think we should keep it.

Reply:
That makes sense. Thanks!